lib/framedivider.rb

```diff
@@ -5,94 +5,216 @@
+    TBD.log(TBD::DEBUG, msg)
     return false
   end
 
-  nom = s.surface.get.nameString
-  parent = model.getSurfaceByName(nom)
-  if parent.empty?
-    TBD.log(TBD::DEBUG,
-      "Can't find subsurface '#{id}' OSM parent surface '#{nom}' - skipping")
+  union = OpenStudio::join(ft_poly1, ft_poly2, TOL)
+  return false if union.empty?
+  union = union.get
+  surface = OpenStudio::Model::Surface.new(union, ft_model)
+  area = surface.grossArea
+  return false if area < TOL
+  if area1 > area2
+    return false if area - area1 > TOL
+  else
+    return false if area - area2 > TOL
+  end
+  true
+end
+
+## Validates whether an OpenStudio polygon overlaps another polygon.
+#
+# @param [Array] poly1 Point3D array of convex polygon #1
+# @param [Array] poly1 Point3D array of convex polygon #2
+# @param [String] id1 Polygon #1 identifier (optional)
+# @param [String] id2 Polygon #2 identifier (optional)
+#
+# @return Returns true if polygons overlaps (or either fits into the other).
+def overlaps?(poly1, poly2, id1 = "", id2 = "")
+  unless poly1 && (poly1.is_a?(OpenStudio::Point3dVector) || poly1.is_a?(Array))
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Invalid polygon (overlaps?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
     return false
   end
+  unless poly2 && (poly2.is_a?(OpenStudio::Point3dVector) || poly2.is_a?(Array))
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Invalid polygon (overlaps?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly1.empty?
+    msg = ""
+    msg << "'#{id1}': " unless id1.empty?
+    msg << "Empty polygon (overlaps?, 1st argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+  if poly2.empty?
+    msg = ""
+    msg << "'#{id2}': " unless id2.empty?
+    msg << "Empty polygon (overlaps?, 2nd argument) - skipping"
+    TBD.log(TBD::DEBUG, msg)
+    return false
+  end
+
+  poly1.each do |pt|
+    unless pt.class == OpenStudio::Point3d
+      msg = "#{pt.class}? "
+      msg << "for '#{id1}': " unless id1.empty?
+      msg << "Expected OpenStudio 3D points (overlaps?, 1st argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return false
+    end
+  end
+  poly2.each do |pt|
+    unless pt.class == OpenStudio::Point3d
+      msg = "#{pt.class}? "
+      msg << "for '#{id2}': " unless id2.empty?
+      msg << "Expected OpenStudio 3D points (overlaps?, 2nd argument - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return false
+    end
+  end
+
+  return true if fits?(poly1, poly2)
+  return true if fits?(poly2, poly1)
+
+  ft = OpenStudio::Transformation::alignFace(poly1).inverse
+  ft_poly1 = flatZ( (ft * poly1).reverse )
+  ft_poly2 = flatZ( (ft * poly2).reverse )
 
-  parent = parent.get
-  ft = OpenStudio::Transformation::alignFace(parent.vertices).inverse
-  ft_parent = (ft * parent.vertices).reverse
-  ft_parent = flatZ(ft_parent)
+  ft_model = OpenStudio::Model::Model.new
```

Comment:
Avoid making a new model and just use the free methods that work on arrays of vertices in https://github.com/NREL/OpenStudio/blob/develop/src/utilities/geometry/Geometry.hpp